src/items/reward.ts

```diff
@@ -37,71 +41,49 @@ import {
 import { TrainerItemEffect, tempStatToTrainerItem } from "#items/trainer-item";
-import {
-  AddPokeballModifier,
-  AddVoucherModifier,
-  EvolutionItemModifier,
-  FusePokemonModifier,
-  type Modifier,
-  MoneyRewardModifier,
-  PokemonAllMovePpRestoreModifier,
-  PokemonHpRestoreModifier,
-  PokemonLevelIncrementModifier,
-  PokemonNatureChangeModifier,
-  PokemonPpRestoreModifier,
-  PokemonPpUpModifier,
-  PokemonStatusHealModifier,
-  RememberMoveModifier,
-  TerrastalizeModifier,
-  TmModifier,
-} from "#modifiers/modifier";
 import type { PokemonMove } from "#moves/pokemon-move";
 import { getVoucherTypeIcon, getVoucherTypeName, VoucherType } from "#system/voucher";
-import type { ModifierTypeFunc, WeightedModifierTypeWeightFunc } from "#types/modifier-types";
+import type { RewardFunc, WeightedRewardWeightFunc } from "#types/rewards";
 import type { PokemonMoveSelectFilter, PokemonSelectFilter } from "#ui/party-ui-handler";
 import { PartyUiHandler } from "#ui/party-ui-handler";
-import { getModifierTierTextTint } from "#ui/text";
-import { formatMoney, isNullOrUndefined, NumberHolder, padInt, randSeedInt, randSeedItem } from "#utils/common";
+import { getRarityTierTextTint } from "#ui/text";
+import {
+  formatMoney,
+  isNullOrUndefined,
+  NumberHolder,
+  padInt,
+  pickWeightedIndex,
+  randSeedInt,
+  randSeedItem,
+} from "#utils/common";
 import { getEnumKeys, getEnumValues } from "#utils/enums";
-import { getModifierPoolForType } from "#utils/modifier-utils";
+import { getRewardPoolForType } from "#utils/reward-utils";
 import i18next from "i18next";
+import { getRewardTierFromPool } from "./init-reward-pools";
+
+export enum RewardClass {
+  REWARD,
+  POKEMON_REWARD,
+  POKEMON_MOVE_REWARD,
+  POKEMON_MOVE_RECALL_REWARD,
+  POKEMON_FUSION_REWARD,
+}
 
-type NewModifierFunc = (type: ModifierType, args: any[]) => Modifier | null;
-
-export class ModifierType {
-  public id: string;
+export class Reward {
+  public id: RewardId;
+  public rewardClass: RewardClass;
   public localeKey: string;
   public iconImage: string;
   public group: string;
   public soundName: string;
-  public tier: RewardTier;
-  protected newModifierFunc: NewModifierFunc | null;
+  public tier: RarityTier;
 
-  /**
-   * Checks if the modifier type is of a specific type
-   * @param modifierType - The type to check against
-   * @return Whether the modifier type is of the specified type
-   */
-  public is<K extends ModifierTypeString>(modifierType: K): this is ModifierTypeInstanceMap[K] {
-    const targetType = ModifierTypeConstructorMap[modifierType];
-    if (!targetType) {
-      return false;
-    }
-    return this instanceof targetType;
-  }
-
-  constructor(
-    localeKey: string | null,
-    iconImage: string | null,
-    newModifierFunc: NewModifierFunc | null,
-    group?: string,
-    soundName?: string,
-  ) {
+  constructor(localeKey: string | null, iconImage: string | null, group?: string, soundName?: string) {
```

Comment:
```suggestion
  constructor(id: RewardId, localeKey: string | null, iconImage: string | null, group?: string, soundName?: string) {
```

Wouldn't it make sense to just pass the RewardIds to the constructor rather than every subclass handling it themselves?